File: app/src/main/java/com/sortedqueue/programmercreek/database/CreekUser.java
package com.sortedqueue.programmercreek.database;

import android.content.Context;

import com.sortedqueue.programmercreek.database.firebase.FirebaseDatabaseHandler;

/**
 * Created by binay on 05/12/16.
 */

public class CreekUser {

    private String emailId;
    private String userFullName;
    private String userPhotoUrl;
    private String programLanguage;
    private String wasAnonUser = "No";
    private String userId = "";


    public CreekUser() {
    }

    public CreekUser(String emailId, String userFullName, String userPhotoUrl, String programLanguage) {
        this.emailId = emailId;
        this.userFullName = userFullName;
        this.userPhotoUrl = userPhotoUrl;
        this.programLanguage = programLanguage;
    }

    public String getWasAnonUser() {
        return wasAnonUser;
    }

    public void setWasAnonUser(String wasAnonUser) {
        this.wasAnonUser = wasAnonUser;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getEmailId() {
        return emailId;
    }

    public void setEmailId(String emailId) {
        this.emailId = emailId;
    }

    public String getUserFullName() {
        return userFullName;
    }

    public void setUserFullName(String userFullName) {
        this.userFullName = userFullName;
    }

    public String getUserPhotoUrl() {
        return userPhotoUrl;
    }

    public void setUserPhotoUrl(String userPhotoUrl) {
        this.userPhotoUrl = userPhotoUrl;
    }

    public String getProgramLanguage() {
        return programLanguage;
    }

    public void setProgramLanguage(String programLanguage) {
        this.programLanguage = programLanguage;
    }

    @Override
    public String toString() {
        return "CreekUser{" +
                "emailId='" + emailId + '\'' +
                ", userFullName='" + userFullName + '\'' +
                ", userPhotoUrl='" + userPhotoUrl + '\'' +
                ", programLanguage='" + programLanguage + '\'' +
                ", wasAnonUser='" + wasAnonUser + '\'' +
                ", userId='" + userId + '\'' +
                '}';
    }

    public void save(Context context) {
        FirebaseDatabaseHandler firebaseDatabaseHandler = new FirebaseDatabaseHandler(context);
        firebaseDatabaseHandler.writeCreekUser(this);
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/database/CreekUser.kt
package com.sortedqueue.programmercreek.database

import android.content.Context
import android.os.Parcel
import android.os.Parcelable

import com.sortedqueue.programmercreek.database.firebase.FirebaseDatabaseHandler

/**
 * Created by binay on 05/12/16.
 */

class CreekUser : Parcelable {

    var emailId: String = ""
    var userFullName: String = ""
    var userPhotoUrl: String = ""
    var programLanguage: String = ""
    var wasAnonUser = "No"
    var userId = ""

    constructor(parcel: Parcel) : this() {
        emailId = parcel.readString() ?: ""
        userFullName = parcel.readString() ?: ""
        userPhotoUrl = parcel.readString()?: ""
        programLanguage = parcel.readString()?: ""
        wasAnonUser = parcel.readString()?: "No"
        userId = parcel.readString()?: ""
    }


    constructor() {}

    constructor(emailId: String, userFullName: String, userPhotoUrl: String, programLanguage: String) {
        this.emailId = emailId
        this.userFullName = userFullName
        this.userPhotoUrl = userPhotoUrl
        this.programLanguage = programLanguage
    }

    override fun toString(): String {
        return "CreekUser{emailId='$emailId', userFullName='$userFullName', userPhotoUrl='$userPhotoUrl', programLanguage='$programLanguage', wasAnonUser='$wasAnonUser', userId='$userId'}"
    }

    fun save(context: Context) {
        val firebaseDatabaseHandler = FirebaseDatabaseHandler(context)
        firebaseDatabaseHandler.writeCreekUser(this)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(emailId)
        parcel.writeString(userFullName)
        parcel.writeString(userPhotoUrl)
        parcel.writeString(programLanguage)
        parcel.writeString(wasAnonUser)
        parcel.writeString(userId)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<CreekUser> {
        override fun createFromParcel(parcel: Parcel): CreekUser {
            return CreekUser(parcel)
        }

        override fun newArray(size: Int): Array<CreekUser?> {
            return arrayOfNulls(size)
        }
    }
}
